src/ProctorioConfig.php

```diff
@@ -43,28 +44,35 @@ class ProctorioConfig
     public const DEFAULT_OAUTH_VERSION = '1.0';
     public const POST_MANHOOD = 'POST';
 
+    /**
+     * @throws ProctorioParameterException
+     */
     public function configure(array $parameters): array
     {
         return [
-            self::LAUNCH_URL => $this->getDefaultValue($parameters, self::LAUNCH_URL, self::getProctorioDefaultUrl()),
-            self::USER_ID => $this->getDefaultValue($parameters, self::USER_ID),
-            self::OAUTH_CONSUMER_KEY => $this->getDefaultValue($parameters, self::OAUTH_CONSUMER_KEY),
-            self::EXAM_START => $this->getDefaultValue($parameters, self::EXAM_START),
-            self::EXAM_TAKE => $this->getDefaultValue($parameters, self::EXAM_TAKE),
+            self::LAUNCH_URL => $parameters[self::LAUNCH_URL],
+            self::USER_ID => $parameters[self::USER_ID],
+            self::OAUTH_CONSUMER_KEY => $parameters[self::OAUTH_CONSUMER_KEY],
+            self::EXAM_START => $parameters[self::LAUNCH_URL],
+            self::EXAM_TAKE => $parameters[self::EXAM_TAKE],
             self::EXAM_END => $this->getDefaultValue($parameters, self::EXAM_END),
             self::EXAM_SETTINGS => $this->getDefaultValue($parameters, self::EXAM_SETTINGS),
-            self::FULL_NAME => $this->getDefaultValue($parameters, self::FULL_NAME),
-            self::EXAM_TAG => $this->getDefaultValue($parameters, self::EXAM_TAG),
+            self::FULL_NAME => $this->getDefaultValue($parameters, self::FULL_NAME, 'name'),
```

Comment:
FULL_NAME and EXAM_TAG are optional and should not be set if they are not provided

Reply:
Is that mean that a value should be `''` or we can generate default value.

Reply:
i think what joel is saying is that if we don't have those values do not create the keys inside the payload 
```suggestion
if(!empty($parameters, self::FULL_NAME)){
            self::FULL_NAME => $this->getDefaultValue($parameters, self::FULL_NAME, 'name'),
            }
```

Reply:
@jbout ? 
@mike-ionut-mihai-sandu-tao from your code suggestion I'm adding if statement to array value assignment.  Additionally, I repeat this same if statement in `getDefaultValue` method, so I'm still confused about what we should send to Proctorio.

Reply:
Exactly, if the user does not provide an EXAM_TAG, we should not add an exam tag, not even an empty one

Reply:
So when we are not including these values in parameters they are gone. Here is a commit: https://github.com/oat-sa/lib-proctorio/pull/5/commits/d3d140d9e86e29fb96eb5884b38b25c1614d440e

Reply:
After a talk with @jbout I made a change to my business logic inside ProctorioConfig in order to provide correct parameters array that we are sending to Proctorio. I created a proper array that will include the correct order of params and information if value have to be sent and default values.  I going to iterate over this array and in this process I'm going to set all required field with provided parameters. 

Please refer to this comment: https://github.com/oat-sa/lib-proctorio/pull/5/commits/3f138160e1de2fb4f27ae7815a62131435fe963d